src/analysis/normed_space/basic.lean

```diff
@@ -316,6 +315,39 @@ nnreal.eq $ by simp
 @[simp] lemma nnnorm_zpow : ∀ (a : α) (n : ℤ), ∥a ^ n∥₊ = ∥a∥₊ ^ n :=
 (nnnorm_hom : α →*₀ ℝ≥0).map_zpow
 
+end normed_division_ring
+
+/-- A normed field is a field with a norm satisfying ∥x y∥ = ∥x∥ ∥y∥. -/
+class normed_field (α : Type*) extends has_norm α, field α, metric_space α :=
+(dist_eq : ∀ x y, dist x y = norm (x - y))
+(norm_mul' : ∀ a b, norm (a * b) = norm a * norm b)
+
+/-- A nondiscrete normed field is a normed field in which there is an element of norm different from
+`0` and `1`. This makes it possible to bring any element arbitrarily close to `0` by multiplication
+by the powers of any element, and thus to relate algebra and topology. -/
+class nondiscrete_normed_field (α : Type*) extends normed_field α :=
+(non_trivial : ∃x:α, 1<∥x∥)
+
+section normed_field
+
+variables [normed_field α]
+
+@[priority 100] -- see Note [lower instance priority]
+instance normed_field.to_normed_division_ring : normed_division_ring α :=
+{ norm_mul' := normed_field.norm_mul', ..‹normed_field α› }
+
+@[priority 100] -- see Note [lower instance priority]
+instance normed_field.to_normed_comm_ring : normed_comm_ring α :=
+{ norm_mul := λ a b, (norm_mul a b).le, ..‹normed_field α› }
+
+@[simp] lemma norm_prod (s : finset β) (f : β → α) :
+  ∥∏ b in s, f b∥ = ∏ b in s, ∥f b∥ :=
+(norm_hom.to_monoid_hom : α →* ℝ).map_prod f s
+
+@[simp] lemma nnnorm_prod (s : finset β) (f : β → α) :
+  ∥∏ b in s, f b∥₊ = ∏ b in s, ∥f b∥₊ :=
+(nnnorm_hom.to_monoid_hom : α →* ℝ≥0).map_prod f s
+
 @[priority 100] -- see Note [lower instance priority]
 instance normed_field.has_continuous_inv₀ : has_continuous_inv₀ α :=
```

Comment:
I'm pretty sure this is easy to generalize to `division_ring`, you just have to use `∥r∥⁻¹ * ∥r - e∥ * ∥e∥⁻¹` or similar below instead of division.

Reply:
yes, done.